crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -337,23 +340,39 @@ impl<'src> Parser<'src> {
         }
     }
 
+    /// Parses an import statement.
+    ///
+    /// # Panics
+    ///
+    /// If the parser isn't positioned at an `import` token.
+    ///
     /// See: <https://docs.python.org/3/reference/simple_stmts.html#grammar-token-python-grammar-import_stmt>
     fn parse_import_statement(&mut self) -> ast::StmtImport {
         let start = self.node_start();
         self.bump(TokenKind::Import);
 
-        let mut names = vec![];
-        #[allow(deprecated)]
-        self.parse_separated(false, TokenKind::Comma, [TokenKind::Newline], |parser| {
-            names.push(parser.parse_alias());
-        });
+        let names = self.parse_comma_separated_list_into_vec(
+            RecoveryContextKind::ImportNames,
+            Parser::parse_alias,
+            // `import a, b,` isn't allowed.
+            false,
+        );
+
+        // TODO(dhruvmanila): Error when `*` is used
```

Comment:
Probably not required as we discussed to move "soft" syntax errors into the linter.